pipeline/mutate/mutator_hydrator.go

```diff
@@ -215,8 +221,21 @@ func (a *MutatorHydrator) Mutate(r *http.Request, session *authn.AuthenticationS
 		return errors.New(ErrNon200ResponseFromAPI)
 	}
 
+	// Handle compressed data
```

Comment:
The go stdlib HTTP client should do that automatically, does it not?

Reply:
From the stdlib HTTP [docs](https://pkg.go.dev/net/http#Transport):

> // If the Transport requests gzip on
	// its own and gets a gzipped response, it's transparently
	// decoded in the Response.Body. However, if the user
	// explicitly requested gzip it is not automatically
	// uncompressed.

So because we set the Accept-Encoding header, we must decompress ourselves. 
Based on what I and Juan131 observed, the transport was not decoding the response prior to that either. 

It's been over 2 years since I touched Go and we've been running this code in production all this time.
However, I have not yet tested again with the latest version of Go.